core/camel-support/src/main/java/org/apache/camel/converter/stream/CipherPair.java

```diff
@@ -61,7 +60,18 @@ public Cipher getEncryptor() {
         return enccipher;
     }
 
-    public Cipher getDecryptor() {
-        return deccipher;
+    /**
+     * Create the decryptor every time because the decryptor is not thead safe. For example, if you reuse the decryptor
+     * instance in the Multi-cast case then you will get errors.
+     */
+    public Cipher createDecryptor() {
+        try {
+            Cipher deccipher = Cipher.getInstance(transformation);
+            deccipher.init(Cipher.DECRYPT_MODE, key, ivp == null ? null : new IvParameterSpec(ivp));
+            return deccipher;
+        } catch (GeneralSecurityException e) {
+            // should not happen
+            throw new IllegalStateException("Could not instantiate decryptor, e");
```

Comment:
There is a little bug here with the caused exception

Reply:
What do you mean with that? Should we not catch the GeneralSecurityException?

Reply:
,e is in the string text, it should be outside

Reply:
We could wrap the GenealSecurityException into an IOException and throw the IOException because this exception is also used in FileInputStreamCache, the only class which uses the createDecryptor method

Reply:
I mean look at the exception message

throw new IllegalStateException("Could not instantiate decryptor, e");

should be

throw new IllegalStateException("Could not instantiate decryptor", e);

Reply:
o.k. I correct that